src/register/cycle.rs

```diff
@@ -0,0 +1,7 @@
+//! cycle register
+//! Shadow of mcycle register
+//! must have mcounteren::cy bit enabled for use in supervisor mode (if implemented)
+//! if supervisor mode is not implemented this register will control user mode access
```

Comment:
```suggestion
//! Must have `scounteren::cy` or `mcounteren::cy` bit enabled depending on whether
//! S-mode is implemented or not.
```
Maybe something like this?